### DISCUSSION TITLE
merge in useObservableState

### DISCUSSION THREAD

**Discussion by BePasquet** [crimx/observable-hooks#61] [Q&A]
Hi, first thanks a lot for creating this library it  has beatifull api, im pretty new to the library and to react i will like to know if there is any way of to merge observables in useObservableState i notice that is emiting multiple time like it was resubscribing every time it change of state an example will be
```
const userImage$ = useObservable((user$) => user$.pipe(pluck('img')), [user]);
const [getMedia, media$] = useObservableCallback<string, DeviceMediaSource>(
  (getMediaClick$) =>
    getMediaClick$.pipe(
      exhaustMap((deviceSource) =>
        deviceSource === DeviceMediaSource.Camera
          ? openCamera()
          : openMediaLibrary()
      )
    )
);

const image = useObservableState(merge(userImage$, media$), '');
```

while this works fine
```
const [image, setImage] = useState('');
const userImage$ = useObservable((user$) => user$.pipe(pluck('img')), [user]);
const [getMedia, media$] = useObservableCallback<string, DeviceMediaSource>(
  (getMediaClick$) =>
    getMediaClick$.pipe(
      exhaustMap((deviceSource) =>
        deviceSource === DeviceMediaSource.Camera
          ? openCamera()
          : openMediaLibrary()
      )
    )
);

useSubscription(userImage$, setImage);
useSubscription(media$, setImage);

```

**Comment by crimx** [comment]
See [docs](https://observable-hooks.js.org/api/#useobservable).

> React function components will be called many times during its life cycle. Create or transform Observables with useObservable so that the operations will not be repeatedly performed.

Try this:

```js
const image$ = useObservable(() => merge(userImage$, media$))
const image = useObservableState(image$, '');
```

`useObservableState` also accepts a similar epic-like callback function. So you can also do something like this:

```js
const [image] = useObservableState(() => merge(userImage$, media$), '');
```

**Reply by BePasquet** [reply]
Thanks a lot for the clarification
